### ISSUE TITLE
When Paying with Expensify the passed payment type is REPORT

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#34904]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

It seems like this is broken since https://github.com/Expensify/App/issues/34621

**Version Number:** v1.4.29-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @joekaufmanexpensify @johnmlee101 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1705929941791079

## Action Performed:
Break down in numbered steps

You should have wallet enabled, for testing purposes I think you can bypass the KYC

1. Submit request to account that already has an upgraded wallet.
1. Click into the Request.
1. Click Pay with Expensify.

## Expected Result:
Describe what you think should've happened

Payment goes through

## Actual Result:
Describe what actually happened

Error appears because the

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by mountiny** [comment]
From the logs we can see that the problem is that in [here](https://github.com/Expensify/App/blob/74a61a2dce36a8bd64758293a88b95297b0d596c/src/libs/actions/IOU.js#L3384) the payment type is `REPORT` for some unclear reason so then the api method called is the `PayMoneyRequest` which is not correct for the Pay with Expensify option

**Comment by mountiny** [comment]
Seems like ti started after this deploy https://github.com/Expensify/App/issues/34621 as only the deploy before is the last one this worked

**Comment by mountiny** [comment]
This is not technically a deploy blocker, but important flow that is not yet covered by regression testing that is broken so trying to fix with urgency

**Comment by mountiny** [comment]
Thanks for linking, yep that will be the exact problem

**Comment by mountiny** [comment]
Most likely will be fixed with this PR https://github.com/Expensify/App/pull/34870

**Comment by mountiny** [comment]
Yeah we would ideally only CP as small change as possible to get this working

**Comment by mountiny** [comment]
@ruben-rebelo @rushatgabhane can you please comment on this issue so I can assign you?

**Comment by mountiny** [comment]
Removing the blocker label as this is technically not a blocker and we have identified the root cause and a fix, currently blocked on something else

**Comment by ruben-rebelo** [comment]
@mountiny sure

**Comment by alexpensify** [comment]
Alright, we are open for proposals here.

**Comment by mountiny** [comment]
This was fixed actually, closing as it was a regression from different PR

**Comment by alexpensify** [comment]
Thank you!
